src/AbService.php

```diff
@@ -11,11 +11,32 @@ class AbService
     protected string $uid;
 
     public function __construct(
+        /** array<int,array<string,array<string,int>>> $experiments */
         protected array $experiments = []
     ) {
+        $this->verifyConfig();
         $this->setUid($this->generateUid());
     }
 
+    public function verifyConfig()
+    {
+        if (empty($this->experiments)) {
+            throw new InvalidArgumentException("No experiments defined");
+        }
+        foreach ($this->experiments as $experiment => $variants) {
+            $sum = 0;
+            foreach ($variants as $variant => $weight) {
+                if (!is_numeric($weight) || $weight < 0) {
+                    throw new InvalidArgumentException("Invalid weight for variant $variant in experiment $experiment");
+                }
+                $sum += $weight;
+            }
+            if ($sum !== 1.0) {
+                throw new InvalidArgumentException("Weights for experiment $experiment do not sum to 1.0");
+            }
+        }
+    }
+
```

Comment:
total waste of resources. This should be done in ab-middleware-bundle during container building.